src/pages/Lend/LendPage.vue

```diff
@@ -25,55 +32,126 @@
 <script>
 import WwwPage from '@/components/WwwFrame/WwwPage';
 import GridLoanCard from '@/components/LoanCards/GridLoanCard';
+import KvLoadingSpinner from '@/components/Kv/KvLoadingSpinner';
+import KvPagination from '@/components/Kv/KvPagination';
 import loanCardQuery from '@/graphql/query/loanCardData.graphql';
 import _get from 'lodash/get';
+import _invokeMap from 'lodash/invokeMap';
 import _map from 'lodash/map';
+// import _mapValues from 'lodash/mapValues';
+import _merge from 'lodash/merge';
+import numeral from 'numeral';
+
+const loansPerPage = 12;
+
+const urlParamTransform = {
+	page: {
+		to({ offset }) {
+			const page = Math.floor(offset / loansPerPage) + 1;
+			return page > 1 ? page : null;
+		},
+		from({ page }) {
+			const pagenum = numeral(page).value() - 1;
+			return { offset: pagenum > 0 ? loansPerPage * pagenum : 0 };
+		}
+	},
+};
+
+// function toUrlParams(variables) {
+// 	return _mapValues(urlParamTransform, ({ to }) => to(variables));
+// }
+
+function fromUrlParams(params) {
+	return _merge({}, ..._invokeMap(urlParamTransform, 'from', params));
+}
 
 export default {
 	components: {
 		WwwPage,
 		GridLoanCard,
+		KvLoadingSpinner,
+		KvPagination,
 	},
 	inject: ['apollo'],
 	metaInfo: {
 		title: 'Fundraising loans'
 	},
 	data() {
 		return {
+			offset: 0,
+			limit: loansPerPage,
 			totalCount: 0,
 			loans: [],
 			isVisitor: true,
 			itemsInBasket: [],
+			loading: false,
 		};
 	},
 	apollo: {
 		query: loanCardQuery,
 		preFetch: true,
+		preFetchVariables({ route }) {
+			return _merge({ limit: loansPerPage }, fromUrlParams(route.query));
+		},
+		variables() {
+			return {
+				offset: this.offset,
+				limit: this.limit,
+			};
+		},
 		result({ data }) {
 			this.totalCount = data.lend.loans.totalCount;
 			this.loans = data.lend.loans.values;
 			this.itemsInBasket = _map(data.shop.basket.items.values, 'id');
 			this.isVisitor = !_get(data, 'my.userAccount.id');
+			this.loading = false;
+		}
+	},
+	watch: {
+		$route(to) {
+			const { offset } = fromUrlParams(to.query);
+			if (this.offset !== offset) {
+				this.loading = true;
+			}
+			this.offset = offset;
 		}
 	}
 };
 </script>
 
 <style lang="scss">
-	@import 'settings';
+@import 'settings';
+
+.lend-page {
+	background-color: $kiva-bg-lightgray;
+
+	.loan-card-group {
+		position: relative;
+	}
+
+	.loading-overlay {
+		position: absolute;
+		background-color: rgba($kiva-bg-lightgray, 0.7);
```

Comment:
Is this valid css? I attempted to copy and paste this in the console for a random element on the page and it won't accept this value: background-color: rgba($kiva-bg-lightgray, 0.7);

Reply:
It's not valid css, but it is valid sass. See http://sass-lang.com/documentation/Sass/Script/Functions.html#rgba-instance_method

From the examples:
```
rgba(#102030, 0.5) => rgba(16, 32, 48, 0.5)
rgba(blue, 0.2)    => rgba(0, 0, 255, 0.2)
```